### PR TITLE
Boosted Range Visuals

### PR DESCRIPTION
**Pull request by MEEPofFaith**
Allows you to see the boosted range of the projectors if you have the materials to boost them unlocked.
Condensed into just one commit.

Co-Authored-By: genNAowl <68400583+genNAowl@users.noreply.github.com>

[Video](https://youtu.be/dLlPSfAGvfc)

### CODE REVIEW

File: core/src/mindustry/world/Block.java [hunk 1]
```diff
@@ -624,6 +624,18 @@ protected void initBuilding(){
...
+    public boolean boosterUnlocked(){
+        boolean unlocked = true;
+        for(ItemStack item : consumes.getItem().items){
+            if(!item.item.unlockedNow() || !consumes.getItem().isBoost()){
+                unlocked = false;
+                break;
+            }
+        }
+        return unlocked;
+    }
```
Comment (summetdev) [inline core/src/mindustry/world/Block.java:638]: Maybe change it like this?

```suggestion
    public boolean boosterUnlocked(){
        for(ItemStack item : consumes.getItem().items){
            if(!item.item.unlockedNow() || !consumes.getItem().isBoost()){
                return false;
            }
        }
        return true;
    }
```

File: core/src/mindustry/world/blocks/defense/MendProjector.java [hunk 2]
```diff
@@ -56,21 +56,13 @@ public void setStats(){
             stats.add(Stat.boostEffect, (phaseBoost + healPercent) / healPercent, StatUnit.timesSpeed);
         }
     }
-
+
```
Comment (Shawak) [inline core/src/mindustry/world/blocks/defense/MendProjector.java:59]: ?

File: core/src/mindustry/world/blocks/defense/OverdriveProjector.java [hunk 3]
```diff
@@ -44,21 +44,13 @@ public OverdriveProjector(String name){
     public boolean outputsItems(){
         return false;
     }
-
+
```
Comment (Shawak) [inline core/src/mindustry/world/blocks/defense/OverdriveProjector.java:47]: ?

File: core/src/mindustry/world/blocks/defense/MendProjector.java [hunk 4]
```diff
@@ -43,7 +43,7 @@ public MendProjector(String name){
     public boolean outputsItems(){
         return false;
     }
-
+
```
Comment (Shawak) [inline core/src/mindustry/world/blocks/defense/MendProjector.java:46]: ?

**Review by Shawak — CHANGES_REQUESTED** [review]

**Comment by MEEPofFaith** [comment]
[Video of new changes](https://youtu.be/xWBVKfIs2ss)

File: core/src/mindustry/world/blocks/defense/MendProjector.java (hunk 4 above)
Comment (MEEPofFaith) [reply core/src/mindustry/world/blocks/defense/MendProjector.java:46]: Dunno why that happened or when I did that.

File: core/src/mindustry/world/blocks/defense/OverdriveProjector.java (hunk 3 above)
Comment (MEEPofFaith) [reply core/src/mindustry/world/blocks/defense/OverdriveProjector.java:47]: Dunno why that happened or when I did that.

File: core/src/mindustry/world/blocks/defense/MendProjector.java (hunk 2 above)
Comment (MEEPofFaith) [reply core/src/mindustry/world/blocks/defense/MendProjector.java:59]: Dunno why that happened or when I did that.

File: gradle.properties [hunk 5]
```diff
@@ -1,3 +1,3 @@
 org.gradle.daemon=true
 org.gradle.jvmargs=-Xms256m -Xmx1024m
-archash=fcc8d5bf44d872ec60eee557f6db16599ed267c2
+archash=46d1f7c89424a7abf0d7c51cd24f2a185ba98337
```
Comment (summetdev) [inline gradle.properties:3]: Why change version of Arc?

Comment (MEEPofFaith) [reply gradle.properties:3]: Dunno, happened on its own. Probably something with Gradle.

Comment (MEEPofFaith) [reply gradle.properties]: And by the power of console, I reverted this change.

File: core/src/mindustry/world/blocks/defense/MendProjector.java [hunk 6]
```diff
@@ -92,9 +120,36 @@ public void updateTile(){
...
+            indexer.eachBlock(this, realRange, other -> true, other -> Drawf.selected(other, baseColor.cpy().lerp(phaseColor, phaseHeat).a(Mathf.absin(4f, 1f))));
 
-            Drawf.dashCircle(x, y, realRange, baseColor);
+            Drawf.dashCircle(x, y, realRange, baseColor.cpy().lerp(phaseColor, phaseHeat));
+            
+            if(!cons().optionalValid() && hasBoost && boosterUnlocked()){
+                indexer.eachBlock(this, range + phaseRangeBoost, other -> true, other -> {
+                  if(!Mathf.within(x, y, other.x, other.y, range)){
+                      Drawf.selected(other, phaseColor.cpy().a(Mathf.absin(4f, 1f)));
+                  }
```
Comment (joshuaptfan) [inline core/src/mindustry/world/blocks/defense/MendProjector.java:131]: Tab size should always be 4 spaces.
```suggestion
                    if(!Mathf.within(x, y, other.x, other.y, range)){
                        Drawf.selected(other, phaseColor.cpy().a(Mathf.absin(4f, 1f)));
                    }
```

**Review by joshuaptfan — CHANGES_REQUESTED** [review]
Style.

**Comment by MEEPofFaith** [comment]
~~I'm sure anuke will be fine with all these little commits.~~

**Comment by MEEPofFaith** [comment]
And the build failed.

**Comment by MEEPofFaith** [comment]
Making more simple version.

**Comment by joshuaptfan** [comment]
Embed the video directly in the PR description this time by uploading to GitHub.

**Pull request closed without merging**